src/sysfiles.c

```diff
@@ -1352,7 +1352,18 @@ Int SyRead(Int fid, void * ptr, size_t len)
         return gzread(syBuf[fid].gzfp, ptr, len);
     }
     else {
+#ifdef EMSCRIPTEN
+        // BUG?: The EMSCRIPTEN library will return "EOF" when
```

Comment:
Does it really return `EOF` (so `-1`) ? From the code it looks more like it returns 0, and that would be valid, I think

Reply:
Aha, I should have first re-read the `read(2)` man page. From https://man7.org/linux/man-pages/man2/read.2.html
> On success, the number of bytes read is returned (zero indicates end of file)

Huh

Reply:
Is the returning of `0` here necessarily an issue in emscripten? Could it also be a problem with xterm.js, or with the code that connects the two?

Reply:
You are right, at first I thought it was returning -1, but it was infact 0.

I'm not positive what piece of code is to blame, I did submit a bug report to xterm-pty.

Reply:
fixed in xterm-pty, so removed this code